### PR TITLE
Drop genericobject.{cpp,h}

### PR DESCRIPTION
**Pull request by nerzhul**
This file is not for generic object but for ActiveObject message passing.
Put ownership of the various commands to the right objects and cleanup the related code.

That should remove some memory overhead due to some extra memory copy on some function, visible on servers with many active objects

Protect m_messages_out from outside modifications

## To do

This PR is Ready for Review.

## How to test

Build and test various active objects.

No function change here

### CODE REVIEW

File: src/content_sao.cpp [hunk 1]
```diff
@@ -289,6 +288,120 @@ void UnitSAO::notifyObjectPropertiesModified()
...
+	// parameters
+	writeS16(os, m_attachment_parent_id);
+	os << serializeString(m_attachment_bone);
+	writeV3F32(os, m_attachment_position);
+	writeV3F32(os, m_attachment_rotation);
+	return os.str();
+}
+
+std::string UnitSAO::generateUpdateBonePositionCommand(const std::string &bone,
+	const v3f &position, const v3f &rotation)
```
Comment (sfan5) [inline src/content_sao.cpp:305]: is there a reason this isn't a const method?

File: src/content_sao.cpp [hunk 2]
```diff
@@ -289,6 +288,120 @@ void UnitSAO::notifyObjectPropertiesModified()
...
+		os<<serializeString(armor_group.first);
+		writeS16(os, armor_group.second);
+	}
+	return os.str();
+}
+
+
+std::string UnitSAO::generateUpdatePositionCommand(const v3f &position, const v3f &velocity,
+	const v3f &acceleration, const v3f &rotation, bool do_interpolate, bool is_movement_end,
+	f32 update_interval)
```
Comment (sfan5) [inline src/content_sao.cpp:358]: same here

File: src/content_sao.cpp (hunk 1 above)
Comment (nerzhul) [reply src/content_sao.cpp:305]: it's static, it cannot be const because there is no membership

File: src/content_sao.cpp (hunk 2 above)
Comment (nerzhul) [reply src/content_sao.cpp:358]: same here :)

**Pull request merged**